### DISCUSSION TITLE
strange error from Platform IO about Audio tools/Vector.h

### DISCUSSION THREAD

**Discussion by SnazzyBlanket** [pschatzmann/arduino-audio-tools#1546] [Q&A] [answered]
I ran the Platform IO code inspector for things effecting performance etc

Oddly it gave a big warning about Line 289 in AudioBasic/Vector.H

CWE-0:Syntax Error: AST broken, binary operator '=' doesn't have two operands.

about this line 

   p_data[len - 1] = T();
   
   in 

`  // removes a single element
  void erase(int pos) {
    if (pos < len) {
      int lenToEnd = len - pos - 1;
      // call destructor on data to be erased
      p_data[pos].~T();
      // shift values by 1 position
      memmove((void *)&p_data[pos], (void *)(&p_data[pos + 1]),
              lenToEnd * sizeof(T));

      // make sure that we have a valid object at the end
      p_data[len - 1] = T();
      len--;
    }
  }

it says it will greatly effect performance. Should I ignore it?

best

**Comment by pschatzmann** [comment, marked as answer]
a) I don't see any performance issue in this call
b) This is currently nowhere used
I committed anyhow a correction since it is possible to call a constructor by indicating the data
